src/sagemaker/model.py

```diff
@@ -1127,7 +1127,7 @@ def _upload_code(self, key_prefix, repack=False):
                 dependencies=self.dependencies,
             )
 
-        if repack:
+        if repack and self.model_data is not None and self.entry_point is not None:
```

Comment:
If we just run estimator.fit followed by estimator.deploy, would this block of code inside the if statement be executed?